src/skmatter/decomposition/_pcov.py

```diff
@@ -0,0 +1,288 @@
+import numbers
+import warnings
+
+import numpy as np
+from numpy.linalg import LinAlgError
+from scipy import linalg
+from scipy.linalg import sqrtm as MatrixSqrt
+from scipy.sparse.linalg import svds
+from sklearn.decomposition._base import _BasePCA
+from sklearn.decomposition._pca import _infer_dimension
+from sklearn.linear_model._base import LinearModel
+from sklearn.utils import check_random_state
+from sklearn.utils._arpack import _init_arpack_v0
+from sklearn.utils.extmath import randomized_svd, stable_cumsum, svd_flip
+from sklearn.utils.validation import check_is_fitted
+
+from skmatter.utils import pcovr_covariance, pcovr_kernel
+
+
+class _BasePCov(_BasePCA, LinearModel):
+    def __init__(
+        self,
+        mixing=0.5,
+        n_components=None,
+        svd_solver="auto",
+        tol=1e-12,
+        space="auto",
+        iterated_power="auto",
+        random_state=None,
+        whiten=False,
```

Comment:
Did we decide if we are starting to use type hints or not?

Reply:
I think we can have a later conversation about type hints, but the general "scikit-learn does not do it" points to no at this time.